src/TopSurface.cpp

```diff
@@ -0,0 +1,63 @@
+//Copyright (c) 2017 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "infill.h"
+#include "LayerPlan.h"
+#include "TopSurface.h"
+
+namespace cura
+{
+
+TopSurface::TopSurface()
```

Comment:
This constructor is unused. You already catch not doing ironing by setting top_surface to nullptr.